### PR TITLE
Extend NavigateToFile selenium test

### PR DESCRIPTION
**Pull request by SkorikSergey**
### What does this PR do?
We need extend **NavigateToFile** selenium test to cover next use cases:

- Closing widget by ESC https://github.com/eclipse/che/issues/6230.
- Widget should not be found files from hidden folders https://github.com/eclipse/che/issues/3928.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/6641

### CODE REVIEW

**Comment by skabashnyuk** [comment]
Pr for branch che6 please.

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/miscellaneous/NavigateToFileTest.java [hunk 1]
```diff
@@ -162,6 +179,32 @@ public void checkNavigateFileFunctionWithJustCreatedFiles() throws Exception {
...
+    // check that HEAD file from .git folder is not appear in list of found files
+    menu.runCommand(
+        TestMenuCommandsConstants.Assistant.ASSISTANT,
+        TestMenuCommandsConstants.Assistant.NAVIGATE_TO_FILE);
+    navigateToFile.waitFormToOpen();
+    loader.waitOnClosed();
+    navigateToFile.typeSymbolInFileNameField("H");
+    loader.waitOnClosed();
+    navigateToFile.waitFileNamePopUp();
+    Assert.assertFalse(navigateToFile.waitListOfFilesNames(FILE_H_SYMBOL));
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/miscellaneous/NavigateToFileTest.java]: Please, use static import to get rid of `Assert.` here to simplify the code.

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/miscellaneous/NavigateToFileTest.java [hunk 2]
```diff
@@ -78,6 +83,9 @@
   @Inject private Menu menu;
   @Inject private TestWorkspaceServiceClient workspaceServiceClient;
   @Inject private TestProjectServiceClient testProjectServiceClient;
+  @Inject private org.eclipse.che.selenium.pageobject.git.Git git;
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/miscellaneous/NavigateToFileTest.java]: Please, move **Git** package to import section to simplify the code.

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/miscellaneous/NavigateToFileTest.java [hunk 3]
```diff
@@ -67,6 +71,7 @@
       Arrays.asList(
           "classpath (/NavigateFile_2/.che)", "classpath (/NavigateFile/.che)",
           "createdFrom.con (/NavigateFile_2)", "createdFrom.api (/NavigateFile)");
+  private static final String FILE_H_SYMBOL = "HEAD (/NavigateFile/.git)";
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/miscellaneous/NavigateToFileTest.java]: Could you, please, add comment or better to rename the constant to explain the exact proposal of it?

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/miscellaneous/NavigateToFileTest.java [hunk 4]
```diff
@@ -162,6 +179,32 @@ public void checkNavigateFileFunctionWithJustCreatedFiles() throws Exception {
...
+        "Do you want to initialize the local repository " + PROJECT_NAME + "?");
+    loader.waitOnClosed();
+    git.waitGitStatusBarWithMess(TestGitConstants.GIT_INITIALIZED_SUCCESS);
+
+    // check that HEAD file from .git folder is not appear in list of found files
+    menu.runCommand(
+        TestMenuCommandsConstants.Assistant.ASSISTANT,
+        TestMenuCommandsConstants.Assistant.NAVIGATE_TO_FILE);
+    navigateToFile.waitFormToOpen();
+    loader.waitOnClosed();
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/miscellaneous/NavigateToFileTest.java]: Waiting of closing of loader looks useless here after the NavigateToFile form had opened, doesn't it?

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/NavigateToFile.java [hunk 5]
```diff
@@ -124,8 +124,8 @@ public void waitFileNamePopUp() {
    *
    * @param text a text that should be into list
    */
-  public void waitListOfFilesNames(final String text) {
-    new WebDriverWait(seleniumWebDriver, LOAD_PAGE_TIMEOUT_SEC)
+  public Boolean waitListOfFilesNames(final String text) {
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/NavigateToFile.java]: "Wait" doesn't require returning the value. IMHO it's better to rename method to `isFilenameSuggested(nameFragment)` according to the actual commands.

**Comment by SkorikSergey** [comment]
Pull request for che6 branch merged https://github.com/eclipse/che/pull/6909.

**Review by dmytro-ndp — APPROVED** [review]

**Pull request merged**